debian/updater.go

```diff
@@ -78,46 +72,30 @@ func (f *Factory) Configure(ctx context.Context, cf driver.ConfigUnmarshaler, c
 		return fmt.Errorf("debian: bad mirror URL: %w", err)
 	}
 
-	u, err = url.Parse(defaultArchive)
-	if cfg.ArchiveURL != "" {
-		u, err = url.Parse(cfg.ArchiveURL)
+	f.json, err = url.Parse(defaultJSON)
+	if cfg.JSONURL != "" {
+		f.json, err = url.Parse(cfg.JSONURL)
 	}
 	if err != nil {
-		return fmt.Errorf("debian: bad archive URL: %w", err)
-	}
-	f.archive, err = u.Parse("debian/")
-	if err != nil {
-		return fmt.Errorf("debian: bad archive URL: %w", err)
-	}
-
-	f.oval, err = url.Parse(defaultOVAL)
-	if cfg.OVALURL != "" {
-		f.oval, err = url.Parse(cfg.OVALURL)
-	}
-	if err != nil {
-		return fmt.Errorf("debian: bad OVAL URL: %w", err)
+		return fmt.Errorf("debian: bad JSON URL: %w", err)
 	}
 
 	return nil
 }
 
 // FactoryConfig is the configuration honored by the Factory.
 //
-// All URLs need trailing slashes.
-//
-// The "archive" and "mirror" URLs expect to find HTML at "dists/" formatted like
+// The "mirror" URLs expect to find HTML at "dists/" formatted like
 // the HTML from the Debian project (that is to say, HTML containing relative links
 // to distribution directories).
 //
-// The "OVAL" URL expects to have OVAL XML documents named "oval-definitions-${name}.xml",
-// where "name" is the release's code name (e.g. "wheezy", "buster").
+// The "mirror" URL needs a trailing slash.
+//
+// The "JSON" URL expects to find a JSON array of packages mapped to related vulnerabilities.
 type FactoryConfig struct {
-	// ArchiveURL is a URL to a Debian archive.
-	ArchiveURL string `json:"archive_url" yaml:"archive_url"`
-	// MirrorURL is a URL to an active Debian mirror.
 	MirrorURL string `json:"mirror_url" yaml:"mirror_url"`
-	// OVALURL is a URL to a collection of OVAL XML documents.
-	OVALURL string `json:"oval_url" yaml:"oval_url"`
+	// JSONURL is a URL to a JSON vulnerability feed.
+	JSONURL string `json:"json_url" yaml:"json_url"`
 }
```

Comment:
The old members should stick around and print warnings if they get used so that a user can figure out why their configuration isn't behaving the same.

Reply:
I was wondering how to handle this "migration". Can we just change it like this or would we need to go through a formal deprecation process?

Reply:
I'm fine with adding some warning prints and a changelog entry. That does mean that the old members still have to stay or be picked up somehow.